streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamKTableJoinProcessor.java

```diff
@@ -49,11 +50,11 @@
     }
 
     @Override
-    public void init(final ProcessorContext context) {
+    public void init(final ProcessorContext<Object, Object> context) {
         super.init(context);
         metrics = (StreamsMetricsImpl) context.metrics();
         droppedRecordsSensor = droppedRecordsSensorOrSkippedRecordsSensor(Thread.currentThread().getName(), context.taskId().toString(), metrics);
-        valueGetter.init(context);
+        valueGetter.init(new ForwardingDisabledProcessorContext(context));
```

Comment:
Even in this first PR, we expose some bugs. In this case, ValueGetters and ValueTransformers shouldn't be able to forward anything, but nothing previously prevented them from doing it. Now, the new bounds force _us_ to wire in a context that has Void bounds.